client/homebrew/navbar/newbrew.navitem.jsx

```diff
@@ -1,12 +1,95 @@
 const React = require('react');
+const { useState, useRef } = React;
+const _ = require('lodash');
 const Nav = require('naturalcrit/nav/nav.jsx');
+const yaml = require('js-yaml');
 
-module.exports = function(props){
-	return <Nav.item
-		href='/new'
-		newTab={true}
-		color='purple'
-		icon='fas fa-plus-square'>
-		new
-	</Nav.item>;
+const BREWKEY  = 'homebrewery-new';
+const STYLEKEY = 'homebrewery-new-style';
+const METAKEY  = 'homebrewery-new-meta';
+
+const NewBrew = () => {
+
+    
+    const [brew, setBrew] = useState({
+        text: '',
+        style: ''
+    });
+
+    const splitTextStyleAndMetadata = (brewContent) => {
```

Comment:
Any chance we can import this function to avoid duplicating it? Maybe extract it from `app.js` into a `/shared/helpers.js` file or something so both the server and the client can use it.

Reply:
helpers sounds extremely vague
what about `fileProcessing.js` or something like that

Reply:
can we do this in a posterior PR? log it in as an issue, then merge this?

Reply:
`helpers` is a pretty standard practice for small functions that need to be shared between multiple areas, and isn't limited to just file processing. `helpers.js` would be a file that potentially contains many small functions that are needed in different places but don't deserve a whole file all to themselves. Alternative names I see a lot are "utilities" or "tools".

> can we do this in a later, separate PR? log it in as an issue, then merge this?

I would rather we just address this now; adding one problem (code duplication) to fix another problem seems like an anti-pattern, especially where the fix is quite easy.